### PR TITLE
feat(obstacle_stop_planner): extend trajectory along vehicle length

### LINKED ISSUE #2513: Autoware ignores obstacles located `vehicle_length` away after goal
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

When an obstacle is in front of the vehicle, on the path:
Screenshot from 2022-12-16 13-46-38

When an obstacle is in front of the vehicle, more than `vehicle_length` away from the goal pose
Screenshot from 2022-12-16 13-47-03

It seems that `obstacle_stop_planner` is not considering the obstacle in the second situation, because the obstacle is further away than `vehicle_length`. I don't know if it is necessary to keep the same safety distance in both situations. However, we definitely need a bigger margin in the second case

### Expected behavior

Keep some safety distance from obstacle, even for the goal.

### Actual behavior

Autoware stops and touches the vehicle in front

### Steps to reproduce

planning simulator

### Versions

_No response_

### Possible causes

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ismetatabay**
## Description

At ```obstacle_stop_planner```, If extend_distance parameter is set to 0.0 by default, ```obstacle_stop_planner``` does not consider the obstacle after vehicle_length. With this PR, extend trajectory by ```max_longitudinal_margin``` parameter for object detection area.
autoware_launch PR: https://github.com/autowarefoundation/autoware_launch/pull/224
## Related links

closes #2513

#### Before This PR :
Initial   |  Arrived 
:-------------------------:|:-------------------------:
 |  

#### After This PR :
Initial   |  Arrived 
:-------------------------:|:-------------------------:
 |  

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/obstacle_stop_planner/src/node.cpp [hunk 1]
```diff
@@ -99,6 +99,8 @@ ObstacleStopPlannerNode::ObstacleStopPlannerNode(const rclcpp::NodeOptions & nod
     p.unknown_lateral_margin =
       declare_parameter<double>(ns + "detection_area.unknown_lateral_margin");
     p.extend_distance = declare_parameter<double>(ns + "detection_area.extend_distance");
+    p.enable_stop_behind_goal_for_obstacle =
+      declare_parameter<bool>(ns + "detection_area.enable_stop_behind_goal_for_obstacle", true);
```
Comment (taikitanaka3) [inline planning/obstacle_stop_planner/src/node.cpp:103]: @ismetatabay please remove default parameter
```suggestion
      declare_parameter<bool>(ns + "detection_area.enable_stop_behind_goal_for_obstacle");
```

Comment (ismetatabay) [reply planning/obstacle_stop_planner/src/node.cpp]: done :+1:

**Comment by ismetatabay** [comment]
@taikitanaka3 Hi, thank you for review. I updated it and i also create PR for autoware_launch parameters.

https://github.com/autowarefoundation/autoware_launch/pull/224

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Pull request merged**